src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -1106,10 +1144,18 @@ rd_kafka_topic_partition_list_t *py_to_c_parts (PyObject *plist) {
 			return NULL;
 		}
 
-		rd_kafka_topic_partition_list_add(c_parts,
+		rd_kafka_topic_partition_t *rktpar;
```

Comment:
All variable definitions needs to go at the top of the scope (e.g., start of function) to be compatible with older compilers.